### PR TITLE
(DEV-3903) Adds Save Button to Prayer List

### PR DESCRIPTION
**Pull request by redreceipt**
## DESCRIPTION
Screen Shot 2019-07-12 at 9 17 05 AM

### What does this PR do, or why is it needed?

This adds a save/unsave button to the prayer list.

### How do I test this PR?

Go in and save/unsave some stuff.
- Check that the UI behaves properly
- Check that it works via the API `savedPrayers` call.

---

> I am affirming this is my _best_ work ([Ecclesiastes 9:10](https://www.bible.com/bible/97/ECC.9.10.MSG))

## TODO

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [x] No new warnings in tests, in storybook, and in-app
- [x] Upload GIF(s) of iOS and Android if applicable
- [x] Set a relevant reviewer

## REVIEW

- [ ] Review updates to test coverage and snapshots
- [ ] Review code through the lens of being concise, simple, and well-documented

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

---

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Comment by richarddubay** [comment]
Screenshot 2019-07-12 10 54 11

@redreceipt The toggle of the button looks great. But when I save a prayer, leave the prayer list and come back to it, the prayer no longer looks like it's saved, even though it really is.

**Comment by redreceipt** [comment]
that's weird, sometimes it works sometimes it doesn't. I was hoping `refetchQueries` would be enough to update cache but maybe it's not good enough

**Comment by redreceipt** [comment]
I *really* trying to avoid using `update` because of the already really ugly super nested-ness of this component but it's not looking good...

**Comment by redreceipt** [comment]
yep, needed `update` the data connection was timing out, explaining why `refetchQueries` wasn't working right.

Ready now 🤞

File: packages/newspringchurchapp/src/prayer/SaveButton/SaveButton.stories.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react-native';
+import { PaddedView } from '@apollosproject/ui-kit';
+
+import SaveButton from '.';
+
+storiesOf('PrayerSingle', module).add('default', () =>
```
Comment (richarddubay) [inline packages/newspringchurchapp/src/prayer/SaveButton/SaveButton.stories.js]:
```suggestion
storiesOf('SaveButton', module).add('default', () =>
```

File: packages/newspringchurchapp/src/prayer/data/mutations/savePrayer.js [hunk 2]
```diff
@@ -4,6 +4,21 @@ export default gql`
   mutation SavePrayer($nodeId: String!) {
     savePrayer(nodeId: $nodeId) {
       id
+      firstName
```
Comment (richarddubay) [inline packages/newspringchurchapp/src/prayer/data/mutations/savePrayer.js:7]: Is all this data necessary? In my mind I would think all you would need to save a prayer would be an Id?

File: packages/newspringchurchapp/src/prayer/data/mutations/unSavePrayer.js [hunk 3]
```diff
@@ -4,6 +4,21 @@ export default gql`
   mutation UnSavePrayer($nodeId: String!) {
     unSavePrayer(nodeId: $nodeId) {
       id
+      firstName
```
Comment (richarddubay) [inline packages/newspringchurchapp/src/prayer/data/mutations/unSavePrayer.js:7]: Same question as the savePrayer mutation.

**Review by richarddubay — COMMENTED** [review]
When I first launched the app and went to the prayer list, the one prayer I had saved did not appear as saved. But once I tapped "Save Prayer" it behaved perfectly after that. So maybe that was just a weird coincidence of caching?

File: packages/newspringchurchapp/src/prayer/data/mutations/savePrayer.js (hunk 2 above)
Comment (redreceipt) [reply packages/newspringchurchapp/src/prayer/data/mutations/savePrayer.js:7]: It's needed for the cache update function of the saved prayers query

**Comment by redreceipt** [comment]
@richarddubay to your first question, good catch. I prob just need a loading state before it knows if it's saved or not, I'll add that in.

File: packages/newspringchurchapp/src/prayer/data/mutations/savePrayer.js (hunk 2 above)
Comment (richarddubay) [reply packages/newspringchurchapp/src/prayer/data/mutations/savePrayer.js:7]: I guess I don't understand the cache update function enough. Just more for me to learn. I'm good with this though.

**Review by richarddubay — APPROVED** [review]

**Pull request merged**